packages/node/src/protocol/utils/get-resolution-increments.ts

```diff
@@ -1,80 +1,74 @@
 import CounterfactualApp from "@counterfactual/contracts/build/CounterfactualApp.json";
-import { AssetType } from "@counterfactual/types";
+import { AssetType, NetworkContext } from "@counterfactual/types";
 import { Contract } from "ethers";
-import { Zero } from "ethers/constants";
+import { One, Zero } from "ethers/constants";
 import { BaseProvider } from "ethers/providers";
-import { BigNumber, bigNumberify } from "ethers/utils";
-import * as log from "loglevel";
+import { BigNumber, bigNumberify, defaultAbiCoder } from "ethers/utils";
 
 import { StateChannel } from "../../models";
 
-type TransferTransaction = {
-  assetType: AssetType;
-  token: string;
-  to: string[];
-  value: BigNumber[];
-  data: string[];
-};
-
 export async function computeFreeBalanceIncrements(
+  networkContext: NetworkContext,
   stateChannel: StateChannel,
   appInstanceId: string,
   provider: BaseProvider
 ): Promise<{ [x: string]: BigNumber }> {
   const appInstance = stateChannel.getAppInstance(appInstanceId);
 
-  const appContract = new Contract(
+  const appDefinition = new Contract(
     appInstance.appInterface.addr,
     CounterfactualApp.abi,
     provider
   );
 
-  let attempts = 1;
-  let resolution: TransferTransaction = await appContract.functions.resolve(
-    appInstance.encodedLatestState,
-    appInstance.terms
+  const resolution = await appDefinition.functions.resolve(
+    appInstance.encodedLatestState
   );
 
-  // FIXME: This retry logic should apply to all view functions _and_ works for
-  //        arbitrary asset types. Presently it only works for ETH resolutions.
-  //        This was added to get the Playground demo launched sooner.
-  const wait = (ms: number) => new Promise(r => setTimeout(r, ms));
-  while (
-    bigNumberify(appInstance.terms.limit).gt(Zero) &&
-    resolution.value.every(v => v.eq(Zero)) &&
-    attempts < 10
-  ) {
-    log.info(`Empty resolution. Querying chain again. Attempt #${attempts}`);
+  /// todo(xuanji)
+  /// there used to be a bunch of retry logic here, to handle the case where
+  /// the appDefinition contract was deployed recently (and hence the provider
+  /// sometimes returning an incorrect definition). This TODO is to handle
+  /// that situation.
```

Comment:
> the appDefinition contract was deployed recently 

No, this is not why the retry logic was here.

The retry logic is to handle the fact that one provider might return one thing (because it observed a balance change that mattered in the `resolve view` function of the ETH balance refund app from the POV of Party A but then Party B might be using a provider where that state has not been observed to have changed yet (it only happened like 1 or 2 blocks ago for example and the chain did not sync yet).

We need to resolve this TODO before merging or the Playground will break.